src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs

```diff
@@ -95,23 +82,59 @@ public class PublishPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
+                string hostType = context.Variables.GetValueOrDefault(WellKnownDistributedTaskVariables.HostType)?.Value; 
+                if (!string.Equals(hostType, "Build", StringComparison.OrdinalIgnoreCase)) {
+                    throw new InvalidOperationException(
+                        StringUtil.Loc("CannotUploadFromCurrentEnvironment", hostType ?? string.Empty)); 
+                }
+
+                if (String.IsNullOrWhiteSpace(artifactName))
+                {
+                    string jobIdentifier = context.Variables.GetValueOrDefault(WellKnownDistributedTaskVariables.JobIdentifier).Value;
+                    var normalizedJobIdentifier = NormalizeJobIdentifier(jobIdentifier);
+                    artifactName = normalizedJobIdentifier;
+                }
+
+                if(!PipelineArtifactPathHelper.IsValidArtifactName(artifactName)) {
+                    throw new ArgumentException(StringUtil.Loc("ArtifactNameIsNotValid", artifactName));
+                }
+
+                string fullPath = Path.GetFullPath(targetPath);
+                bool isFile = File.Exists(fullPath);
+                bool isDir = Directory.Exists(fullPath);
+                if (!isFile && !isDir)
+                {
+                    // if local path is neither file nor folder
+                    throw new FileNotFoundException(StringUtil.Loc("PathDoesNotExist", targetPath));
+                }
+
+                // Upload to VSTS BlobStore, and associate the artifact with the build.
+                context.Output(StringUtil.Loc("UploadingPipelineArtifact", fullPath, buildId));
+                PipelineArtifactServer server = new PipelineArtifactServer();
+                await server.UploadAsync(context, projectId, buildId, artifactName, fullPath, token);
+                context.Output(StringUtil.Loc("UploadArtifactFinished"));
+
             }
+            else if (artifactType == fileShareType)
+            {
+                string fileSharePath = context.GetInput(ArtifactEventProperties.FileSharePath, required: true);
 
-            // Upload to VSTS BlobStore, and associate the artifact with the build.
-            context.Output(StringUtil.Loc("UploadingPipelineArtifact", fullPath, buildId));
-            PipelineArtifactServer server = new PipelineArtifactServer();
-            await server.UploadAsync(context, projectId, buildId, artifactName, fullPath, token);
-            context.Output(StringUtil.Loc("UploadArtifactFinished"));
+                if(RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    FilePathUNCServer server = new FilePathUNCServer();
+                    await server.UploadAsync(context, projectId, buildId, artifactName, targetPath, fileSharePath, token);                  
+                }
+                else 
+                {
+                    // file share artifacts are not currently supported on OSX/Linux.
```

Comment:
Is this true of the existing task?

Reply:
Yes this comment and the behavior is as the same as the FileShare in the build artifact task.